### DISCUSSION TITLE
Open up postgres `TransactionBuilder::new` to `pub` visibility

### DISCUSSION THREAD

**Discussion by evforde** [diesel-rs/diesel#3500] [Ideas]
Hey folks!

> - What do you want to do, and how do you expect Diesel to support you with that?
> 
> I'm looking to implement a drop-in replacement for `diesel::pg::PgConnection` that instruments various methods. The WIP is [here](https://github.com/CQCL/diesel-tracing/blob/feature/diesel-2.0/src/pg.rs#L32). This includes implementing `SimpleConnection`, `Connection`, and [`build_transaction`](https://github.com/diesel-rs/diesel/blob/v2.0.3/diesel/src/pg/connection/mod.rs#L328), (which isn't attached to any trait).
> 
> - How might this be added to Diesel?
> 
> My goal is currently made slightly difficult because [`TransactionBuilder::new`](https://github.com/diesel-rs/diesel/blob/v2.0.3/diesel/src/pg/transaction.rs#L32) has `pub(crate)` visibility. In order to re-implement `build_transaction()` for my drop-in replacement for `PgConnection`, I'd need to also rewrite the `TransactionBuilder`. But, it seems `TransactionBuilder<C>` would work fine with any `C: Connection<Backend = Pg, TransactionManager = AnsiTransactionManager>` and could be reused in my case - do correct me if I'm wrong here, I'm new 😃 
> 
> - What are possible alternatives?
> 
> Re-implementing `TransactionBuilder` outside of diesel for my purpose.
> 
> - Are there any disadvantages?
> 
> Not that I can tell... but correct me if I am missing something.
> 

Thanks for your help, appreciate it!

**Comment by weiznich** [comment]
I would happily accept a PR exposing this function behind the `i-implement-a-third-party-backend-and-opt-into-breaking-changes` feature flag. It seems like we just missed exposing it as part of the diesel 2.0 release.

It would be great if you submit a PR that just adds to the function signature
```rust
    #[diesel_derives::__diesel_public_if(
        feature = "i-implement-a-third-party-backend-and-opt-into-breaking-changes"
    )]
```
Otherwise I will try to take care of that in the next weeks before the next minor diesel release.
